components/org.wso2.carbon.identity.keyrotation/src/main/java/org.wso2.carbon.identity.keyrotation/config/KeyRotationConfig.java

```diff
@@ -0,0 +1,424 @@
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.keyrotation.config;
+
+import org.apache.axiom.om.util.Base64;
+import org.apache.log4j.Logger;
+import org.wso2.carbon.identity.keyrotation.util.KeyRotationException;
+import org.wso2.carbon.identity.keyrotation.util.KeyRotationServiceUtils;
+
+import java.io.File;
+import java.lang.reflect.Field;
+import java.nio.file.Paths;
+
+/**
+ * This class holds implementations needed to load the configurations in the properties.yaml file.
+ */
+public class KeyRotationConfig {
+
+    private String oldSecretKey;
+    private String newSecretKey;
+    private String newISHome;
+    private String oldIdnDBUrl;
+    private String oldIdnUsername;
+    private String oldIdnPassword;
+    private String newIdnDBUrl;
+    private String newIdnUsername;
+    private String newIdnPassword;
+    private String newRegDBUrl;
+    private String newRegUsername;
+    private String newRegPassword;
+    private String enableDBMigrator;
+    private String enableConfigMigrator;
+    private String enableSyncMigrator;
+    private static final Logger log = Logger.getLogger(KeyRotationConfig.class);
+    private static final KeyRotationConfig instance = new KeyRotationConfig();
+
+    public static KeyRotationConfig getInstance() {
+
+        return instance;
+    }
+
+    /**
+     * Load the configurations placed in the properties.yaml file.
+     *
+     * @return KeyRotation config object.
+     * @throws KeyRotationException Exception thrown if something unexpected happens during key rotation.
+     */
+    public KeyRotationConfig loadConfigs(String[] args) throws KeyRotationException {
+
+        String propertiesFilePath = Paths.get("components", "org.wso2.carbon.identity.keyrotation", "src",
+                "main", "resources", "properties.yaml").toString();
+        if (args.length > 0) {
+            propertiesFilePath = args[0];
+        }
+        File file = new File(propertiesFilePath);
+        if (!file.exists()) {
+            throw new KeyRotationException(
+                    "Error occurred, properties.yaml file not found in provided path, " + propertiesFilePath);
+        }
+        log.debug("Loading Key Rotation Configs from path: " + propertiesFilePath);
+        KeyRotationConfig keyRotationConfig =
+                KeyRotationServiceUtils.loadKeyRotationConfig(propertiesFilePath);
+        checkKeyRotationConfigs(keyRotationConfig);
+        log.debug("Successfully loaded the config file.");
```

Comment:
Shall we add an info log? WDYT?

Reply:
Sure,  addressed with https://github.com/wso2/identity-tools/pull/13/commits/6b6184cfde572cff05cdeefedb6e9233fd11d0e8